systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/ListenersST.java

```diff
@@ -389,6 +389,90 @@ void testSendMessagesTlsScramSha(ExtensionContext extensionContext) {
         assertThat(serviceDiscoveryArray, is(StUtils.expectedServiceDiscoveryInfo(9096, "kafka", "scram-sha-512", true)));
     }
 
+    /**
+     * Test custom listener configured with scram sha auth and tls.
+     */
+    @ParallelNamespaceTest
+    @Tag(ACCEPTANCE)
+    @Tag(INTERNAL_CLIENTS_USED)
+    void testSendMessagesCustomListenerTlsScramSha(ExtensionContext extensionContext) {
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(INFRA_NAMESPACE, extensionContext);
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String topicName = mapWithTestTopics.get(extensionContext.getDisplayName());
+        final String kafkaUsername = mapWithTestUsers.get(extensionContext.getDisplayName());
+        final int passwordLength = 25;
+
+        // Use a Kafka with plain listener disabled
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3)
+                .editSpec()
+                .editKafka()
+                .withListeners(new GenericKafkaListenerBuilder()
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                        .withPort(9122)
+                        .withTls(true)
+                        .withNewKafkaListenerAuthenticationCustomAuth()
+                        .withSasl(true)
+                        .withListenerConfig(Map.of("scram-sha-512.sasl.jaas.config",
+                                "org.apache.kafka.common.security.scram.ScramLoginModule required;",
+                                "sasl.enabled.mechanisms", "SCRAM-SHA-512"))
+                        .endKafkaListenerAuthenticationCustomAuth()
+                        .build())
+                .endKafka()
+                .editEntityOperator()
+                .editOrNewTemplate()
+                .editOrNewUserOperatorContainer()
+                .addToEnv(
+                        new ContainerEnvVarBuilder()
+                                .withName("STRIMZI_SCRAM_SHA_PASSWORD_LENGTH")
+                                .withValue(String.valueOf(passwordLength))
+                                .build())
+                .endUserOperatorContainer()
+                .endTemplate()
+                .endEntityOperator()
```

Comment:
I thought this would be needed for
 ```        internalKafkaClient.checkProducedAndConsumedMessages(
                internalKafkaClient.sendMessagesTls(),
                internalKafkaClient.receiveMessagesTls()
        );
```